docs/decisions/0007-handling-learner-initiated-unenrollments.rst

```diff
@@ -0,0 +1,101 @@
+  transaction.
+
+Consequences    
+************
+
+Event bus driven architecture vs routine jobs and active monitoring
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+This management command oriented approach is not ideal as it is not event bus driven. However, the evaluation work done
+on the part of the subsidy service can easily be bootstrapped by future event bus driven implementations. The hope is
+that future iterations on the enterprise subsidy service will be able to hook up an event receiver to the same methods
+run by the management command, but instead of manually fetching and evaluating recently unenrolled records, it would 
+evaluate objects which have triggered an enterprise fulfillment unenrollment/cancellation event. 
+
+Introducing the subsidy service to Jenkins
+++++++++++++++++++++++++++++++++++++++++++
+
+As it currently stands (May 2021), the enterprise subsidy service is not a part of the Jenkins CI pipeline. The
+accepted implementation of running routine jobs for our services hosted on kubernetes is to utilize argoCD. within
+our internal configurations we can ensure that the job is run on a schedule of our choosing.
+
+Admin initiated unenrollment DAG and introducing a transaction reversal writing task
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+The subsidy service currently utilizes specific event driven architecture to handle admin initiated unenrollments.
+This is done by listening for reversal object creations and then making a call to platform to unenroll all associated
+fulfillment objects. This runs in opposition to the planned management command which will write reversals based on the 
+cancellation of fulfillment objects. In order to mitigate any cyclical behaviors between the two systems, we will need
+to implement two stop-gaps.
+
+- The platform enterprise fulfillment cancellation endpoint needs to verify if the fulfillment object is already
+  revoked and ensure that the unenrolled_at time of the course enrollment object is not bumped if so.
+
+- The subsidy service must check if a recently unenrolled fulfillment object already has an associated reversal object.
+
+How the systems will interact with one another
+++++++++++++++++++++++++++++++++++++++++++++++
+
+Records updated and monitored are limited to the subsidy and edx platform services. Notably, platform course enrollment
+and enterprise fulfillment objects are both housed within edx platform but are split between the enterprise package and
+the LMS. The subsidy service will be responsible for maintaining reversals, as well as their transaction counterparts.
+
+The flow for learner initiated unenrollments will be as follows:
+
+1. Learners unenroll from a course. Notably this has multiple origin sources as it can be done from the LMS or from
+   the enterprise learner/admin dashboards, however the end result is that the course enrollment object on platform is
```

Comment:
I might refer to it as the `student.CourseEnrollment` object to help disambiguate.  Also maybe "de-activated" instead of "cancelled" to jibe with the field name.